api/src/main/java/org/apache/iceberg/AppendFiles.java

```diff
@@ -42,16 +42,17 @@ public interface AppendFiles extends SnapshotUpdate<AppendFiles> {
    * <p>The manifest must contain only appended files. All files in the manifest will be appended to
    * the table in the snapshot created by this update.
    *
-   * <p>By default, the manifest will be rewritten to assign all entries this update's snapshot ID.
-   * In that case, it is always the responsibility of the caller to manage the lifecycle of the
-   * original manifest.
+   * <p>The manifest will be used directly if snapshot ID inheritance is enabled (all tables with
+   * the format version &gt; 1 or if the inheritance is enabled explicitly via table properties).
+   * Otherwise, the manifest will be rewritten to assign all entries this update's snapshot ID.
    *
-   * <p>If manifest entries are allowed to inherit the snapshot ID assigned on commit, the manifest
-   * should never be deleted manually if the commit succeeds as it will become part of the table
-   * metadata and will be cleaned up on expiry. If the manifest gets merged with others while
-   * preparing a new snapshot, it will be deleted automatically if this operation is successful. If
-   * the commit fails, the manifest will never be deleted and it is up to the caller whether to
-   * delete or reuse it.
+   * <p>If the manifest is rewritten, it is always the responsibility of the caller to manage the
+   * lifecycle of the original manifest. If manifest entries are allowed to inherit the snapshot ID
```

Comment:
What do you think to simplify 'if manifest entries are allowed to inherit the snapshot ID...' to 'if the manifest is used directly'..?

Reply:
Yep, I liked it. Fixed.